WooCommerce/Classes/ViewRelated/Dashboard/JetpackInstall/JetpackInstallStepsViewModel.swift

```diff
@@ -17,40 +17,75 @@ final class JetpackInstallStepsViewModel: ObservableObject {
     ///
     @Published private(set) var currentStep: JetpackInstallStep?
 
+    /// Whether the install failed. This will be observed by `JetpackInstallStepsView` to present error modal.
+    ///
+    @Published var installFailed: Bool = false
+
     init(siteID: Int64, stores: StoresManager = ServiceLocator.stores) {
         self.siteID = siteID
         self.stores = stores
     }
 
-    /// Starts the steps by installing the Jetpack plugin.
+    /// Starts the steps by checking Jetpack-the-plugin.
     ///
     func startInstallation() {
+        checkJetpackPluginDetailsAndProceed()
+    }
+
+    /// Fetches details for Jetpack-the-plugin, and installs it if the plugin does not exist.
+    /// Otherwise proceeds to activate the plugin if needed.
+    /// - Parameters:
+    ///   - retryCount: number of retries done for error handling.
+    ///
+    private func checkJetpackPluginDetailsAndProceed(retryCount: Int = 0) {
+        guard retryCount <= Constants.maxRetryCount else {
+            installFailed = true
+            return
+        }
+        let pluginInfoAction = SitePluginAction.getPluginDetails(siteID: siteID, pluginName: Constants.jetpackPluginName) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let plugin):
+                if plugin.status == .active {
+                    self.checkSiteConnection()
+                } else {
+                    self.activateJetpack(retryCount: retryCount)
+                }
+            case .failure:
+                // plugin is likely to not have been installed, so proceed to install it.
+                self.installJetpackPlugin(retryCount: retryCount)
+            }
+        }
+        stores.dispatch(pluginInfoAction)
+    }
+
+    /// Installs Jetpack plugin to current site.
+    ///
+    private func installJetpackPlugin(retryCount: Int = 0) {
         currentStep = .installation
         let installationAction = SitePluginAction.installSitePlugin(siteID: siteID, slug: Constants.jetpackSlug) { [weak self] result in
             guard let self = self else { return }
             switch result {
             case .success:
                 self.activateJetpack()
```

Comment:
I was thinking that the activate call here is "new", so it shouldn't belong to the "retry chain" of the installation step. But then I see that the call in `checkJetpackPluginDetailsAndProceed` does get the `retryCount` when the installation step is retried, so it gets messy there. WDYT is the preferred solution here?

Reply:
It makes sense to reset the retry count for each installation and activation. How about we keep `retryCount` as an internal variable and reset it before proceeding to the next step? So that we allow the max number of retries for each step